homeassistant/components/iaqualink/climate.py

```diff
@@ -0,0 +1,156 @@
+"""Support for Aqualink Thermostats."""
+import logging
+from typing import TYPE_CHECKING, List, Optional
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    DOMAIN,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN as AQUALINK_DOMAIN, CLIMATE_SUPPORTED_MODES
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+if TYPE_CHECKING:
+    from iaqualink import (
+        AqualinkHeater,
+        AqualinkPump,
+        AqualinkSensor,
+        AqualinkThermostat,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up discovered switches."""
+    devs = []
+    for dev in hass.data[AQUALINK_DOMAIN][DOMAIN]:
+        devs.append(HassAqualinkThermostat(dev))
+    async_add_entities(devs, True)
+
+
+class HassAqualinkThermostat(ClimateDevice):
+    """Representation of a thermostat."""
+
+    def __init__(self, dev: "AqualinkThermostat"):
+        """Initialize the thermostat."""
+        self.dev = dev
+
+    @property
+    def name(self) -> str:
+        """Return the name of the thermostat."""
+        return self.dev.label.split(" ")[0]
+
+    async def async_update(self) -> None:
+        """Update the internal state of the thermostat.
+
+        The API update() command refreshes the state of all devices so we
+        only update if this is the main thermostat to avoid unnecessary
+        calls.
+        """
+        if self.name != "Pool":
+            return
+        await self.dev.system.update()
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of supported HVAC modes."""
+        return CLIMATE_SUPPORTED_MODES
+
+    @property
+    def pump(self) -> "AqualinkPump":
+        """Return the pump device for the current thermostat."""
+        pump = self.name.lower() + "_pump"
+        return self.dev.system.devices[pump]
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current HVAC mode."""
+        from iaqualink import AqualinkState
```

Comment:
Please move 3rd party imports to the top of the module.

Reply:
SGTM